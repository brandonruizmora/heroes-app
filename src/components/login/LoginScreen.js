import React from 'react';
import { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { useForm } from '../../hooks/useForm';
import { types } from '../../types/types';
import '../../login.css';

export const LoginScreen = ({ history }) => {

    const { dispatch } = useContext(AuthContext);

    const [{ userName }, handleInputChange, reset] = useForm({ userName: '' });

    const handleLogin = () => {

        const lastPath = localStorage.getItem('lastPath') || '/';

        let userNameLogin = 'anonymous';

        if (!userName.length <= 0) {
            userNameLogin = userName;
        }

        dispatch({
            type: types.login,
            payload: {
                name: userNameLogin
            }
        });
        reset();
        history.replace(lastPath);
    }

    return (
        <div className="container">
            <div className="fill d-flex align-items-center justify-content-center">
                <div className="card w-50 h-50">
                    <div className="card-header">
                        <h1>Login Screen</h1>
                    </div>
                    <div className="card-body d-flex align-items-center justify-content-center">
                        <div className="row">
                            <div className="col-12">
                                <input
                                    type="text"
                                    className="form-control"
                                    autoComplete="off"
                                    name="userName"
                                    placeholder="Write an username..."
                                    onChange={handleInputChange}
                                    value={userName}
                                    maxLength="20"
                                />
                                <small className="fw-light fst-italic float-end me-1">{userName.length} / 20</small>
                            </div>
                            <div className="col-12">
                                <div className="d-grid gap-1 w-100">
                                    <button
                                        className="btn btn-dark mt-4"
                                        onClick={handleLogin}
                                    >Login</button>

                                </div>
                            </div>
                        </div>
                    </div>
                </div>

            </div>
        </div>
    )
}
