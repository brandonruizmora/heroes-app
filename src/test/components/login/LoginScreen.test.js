import '../../../setupTests';
import { shallow, mount } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Pruebas en <LoginScreen />', () => {

    const history = {
        replace: jest.fn()
    };

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={history} />
        </AuthContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de realizar el dispatch y la navegacion', () => {
        wrapper.find('button').simulate('click');
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'anonymous'
            }
        });
        expect(history.replace).toHaveBeenCalledTimes(1);
        localStorage.setItem('lastPath', '/dc');
        wrapper.find('button').simulate('click');
        expect(history.replace).toHaveBeenCalledWith('/dc');
    });
    


});
