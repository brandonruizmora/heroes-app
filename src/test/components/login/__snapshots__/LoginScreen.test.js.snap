// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe de mostrarse correctamente 1`] = `
<div
  className="container"
>
  <div
    className="fill d-flex align-items-center justify-content-center"
  >
    <div
      className="card w-50 h-50"
    >
      <div
        className="card-header"
      >
        <h1>
          Login Screen
        </h1>
      </div>
      <div
        className="card-body d-flex align-items-center justify-content-center"
      >
        <div
          className="row"
        >
          <div
            className="col-12"
          >
            <input
              autoComplete="off"
              className="form-control"
              maxLength="20"
              name="userName"
              onChange={[Function]}
              placeholder="Write an username..."
              type="text"
              value=""
            />
            <small
              className="fw-light fst-italic float-end me-1"
            >
              0
               / 20
            </small>
          </div>
          <div
            className="col-12"
          >
            <div
              className="d-grid gap-1 w-100"
            >
              <button
                className="btn btn-dark mt-4"
                onClick={[Function]}
              >
                Login
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
